v2/internal/controllers/crd_datafactory_test.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	datafactory "github.com/Azure/azure-service-operator/v2/api/datafactory/v1api20180601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+func Test_Data_Factory_CRUD(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Create a data factory instance
+	factory := &datafactory.Factory{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("datafactory")),
+		Spec: datafactory.Factory_Spec{
```

Comment:
Suggest including some values for `GlobalProperties` to ensure those round-trip correctly (since that's a map of names and values rather than something strongly typed)

Reply:
Added GlobalParameters